lib/streamlit/watcher/event_based_path_watcher.py

```diff
@@ -339,19 +342,25 @@ def handle_path_change_event(self, event: events.FileSystemEvent) -> None:
             _LOGGER.debug("Don't care about event type %s", event.event_type)
             return
 
-        changed_path = os.path.abspath(changed_path)
+        # Watchdog 5.X is supported Python >=3.9, so watchdog 4.X is used for Python 3.8.
+        # In Watchdog 5.X, the path can be bytes or str, but in Watchdog 4.X, the path is always str,
+        # that's why we convert the path to str, but we need to ignore the unreachable code warning for Python 3.8.
+        if isinstance(changed_path, bytes):  # type: ignore[unreachable, unused-ignore]
```

Comment:
I have just realized that we should have an additional test for this new scenario here in the `event_based_path_watcher_test.py` file

Reply:
I added the test! :)

I am not sure if it will ever pass a bytes object in practice, but it's good to have a test for it.

Reply:
That's awesome, thank you! 🙂 🚀